### PR TITLE
HOSTEDCP-2193: reuse SPs and MIs in hypershift aks e2e

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**

### CODE REVIEW

File: ci-operator/step-registry/hypershift/azure/aks/attach-kv/hypershift-azure-aks-attach-kv-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ref:
+  as: hypershift-azure-aks-attach-kv
+  from_image:
+    namespace: ocp
+    name: "4.16"
```
Comment (bryan-cox) [inline ci-operator/step-registry/hypershift/azure/aks/attach-kv/hypershift-azure-aks-attach-kv-ref.yaml:5]: Should this be updated?

**Comment by bryan-cox** [comment]
Can you add descriptions to the commits on the "why" please?

**Comment by bryan-cox** [comment]
/retitle HOSTEDCP-2193: reuse SPs and MIs in hypershift aks e2e

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
@jianlinliu could you ptal and approve the change in step-registry/aks/provision/aks-provision-commands.sh

**Comment by jianlinliu** [comment]
> @jianlinliu could you ptal and approve the change in step-registry/aks/provision/aks-provision-commands.sh

@fxierh Can you take a look at that ?

**Comment by bryan-cox** [comment]
@jianlinliu - @fxierh is no longer with Red Hat.

**Comment by heliubj18** [comment]
@jianlinliu I will take the aks provision steps for hypershift related reviews once https://github.com/openshift/release/pull/59724 is merged

**Pull request merged**